src/relay/transforms/annotate_target.cc

```diff
@@ -161,8 +200,9 @@ class AnnotateTargetRewriter : public ExprRewriter {
       const CallNode* first_arg_call = pre->args[0].as<CallNode>();
       if (first_arg_call && first_arg_call->op == CompilerBeginOp()) {
         std::string arg_target = first_arg_call->attrs.as<CompilerAttrs>()->compiler;
-        if (arg_target != "default") {
-          supported_targets.push_back(arg_target);
+        if (arg_target != default_target) {
+          // annotated already
+          return post;
```

Comment:
Looks like you remove the feature that considers the target in existing annotation nodes?

Reply:
Here it is peeking first arg and if it is already annotated with non-default target it returns the node untouched, preserving the target.